src/controllers/baseController.ts

```diff
@@ -0,0 +1,11 @@
+import { Response } from 'express';
+import { constants } from '../constants/constants.js';
+import { User } from '../types/user.js';
+
+
+export class BaseController {
```

Comment:
good idea to create and incapsulate duplicating logic into the `BaseController`. But from what I can see, it is not totally the base one as you use `User` object as parameter of the method. 

Let's imagine, tomorrow, I will get requirements to add a few more models, like `Product`, 'Invoice' and `Document`.  So, base on that logic, you implemented, you need to add either new `BaseController` or extend/overload the method `success` in order to support new types. Well, the both, mentioned solutions are not suitable enough for our case.

What I would recommend is to create custom `interface` with the predefined structure like:

```
export class SerialisedModel {
   constructor() ...
}
...

class UseModel extends SerialisedModel {}

...

success(res: Response, data: SerialisedModel): Response {

```